Google.Ads.GoogleAds/examples/AdvancedOperations/AddDemandGenCampaign.cs

```diff
@@ -0,0 +1,485 @@
+                Console.WriteLine($"Message: {e.Message}");
+                Console.WriteLine($"Failure: {e.Failure}");
+                Console.WriteLine($"Request ID: {e.RequestId}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Creates a MutateOperation that creates a new CampaignBudget.
+        /// A temporary ID will be assigned to this campaign budget so that it can be referenced by
+        /// other objects being created in the same Mutate request.
+        /// </summary>
+        /// <param name="budgetResourceName">The temporary resource name of the budget to
+        /// create.</param>
+        /// <returns>A MutateOperation that creates a CampaignBudget</returns>
+        private MutateOperation CreateCampaignBudgetOperation(string budgetResourceName)
+        {
+            return new MutateOperation
+            {
+                CampaignBudgetOperation = new CampaignBudgetOperation
+                {
+                    Create = new CampaignBudget
+                    {
+                        Name = $"Demand Gen campaign budget {ExampleUtilities.GetRandomString()}",
+
+                        // The budget period already defaults to DAILY.
+                        AmountMicros = 50000000,
+
+                        // A Demand Gen campaign cannot use a shared campaign budget.
+                        ExplicitlyShared = false,
+
+                        // Set a temporary ID in the budget's resource name so it can be referenced
+                        // by the campaign in later steps.
+                        ResourceName = budgetResourceName
+                    }
+                }
+            };
+        }
+
+        /// <summary>
+        /// Creates a MutateOperation that creates a new Campaign.
+        /// A temporary ID will be assigned to this campaign so that it can be referenced by
+        /// other objects being created in the same Mutate request.
+        /// </summary>
+        /// <param name="campaignResourceName">The temporary resource name of the campaign to
+        /// create.</param>
+        /// <param name="budgetResourceName">The temporary resource name of the budget to
+        /// assign to the campaign.</param>
+        /// <returns>A MutateOperation that creates a Campaign</returns>
+        // [START add_demand_gen_campaign_2]
+        private MutateOperation CreateDemandGenCampaignOperation(
+            string campaignResourceName, string budgetResourceName)
+        {
+            return new MutateOperation
+            {
+                CampaignOperation = new CampaignOperation
+                {
+                    Create = new Campaign
+                    {
+                        Name = $"Demand Gen #{ExampleUtilities.GetRandomString()}",
+                        // Set the campaign status as PAUSED. The campaign is the only entity in
+                        // the mutate request that should have its status set.
```

Comment:
Note that in `CreateAdGroupOperation` the ad group's status is set to `ENABLED` so there is contradicting information. Should we update this comment, or remove the line that sets the ad groups status?

Reply:
Right, fixed in #602.